### ISSUE TITLE
Integrate data account S3 events to OrcaBus event source SQS

### ISSUE THREAD

**Issue by victorskl** [umccr/infrastructure#434]
**Context:**

Find a solution to send S3 events from data account to beta(dev)|gamma(stg)|prod.

Couple of ways possible with EventBridge to EventBridge cross accounts integration.

See
https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-cross-account.html

**Actions:**

- Integrate it with current data account bucket setup
- Route data account S3 events to [OrcaBus event-source SQS target](https://github.com/umccr/orcabus/blob/aff84b1b7518b221e4159cf6032056c85b81fc05/lib/workload/stateful/stacks/shared/stack.ts#L66-L68)

**Comment by victorskl** [comment]
This integration task in the critical path for OrcaBus release Phase 1.

**Comment by victorskl** [comment]
Amazon EventBridge - Routing events to buses in other AWS accounts
https://www.youtube.com/watch?v=pX_xIW_EuCE

**Comment by victorskl** [comment]
Simplifying cross-account access with Amazon EventBridge resource policies
https://aws.amazon.com/blogs/compute/simplifying-cross-account-access-with-amazon-eventbridge-resource-policies/

**Comment by reisingerf** [comment]
Started working on this on the [feature/data-s3-event-forward](https://github.com/umccr/infrastructure/tree/feature/data-s3-event-forward) branch

**Comment by reisingerf** [comment]
This will need a resource policy counterpart on the target event bus(es) to allow PutEvents from the source (data) account
=> OrcaBus CDK update

**Comment by reisingerf** [comment]
Changes to OrcaBus event bus are on the [update/x-account-events](https://github.com/umccr/orcabus/tree/update/x-account-events) branch of the orcabus repo.

**Comment by victorskl** [comment]
Let us pair coding and complete this together tomorrow, Flo.

**Comment by victorskl** [comment]
Caveat 1:

- When we integrate our BYOB bucket to ICAv2 project as underlay storage configuration, ICAv2 subscribes couple of bucket notifications to our buckets.
- These bucket notification subscription are not in our control.
- When we also want to integrate our bucket notification to EventBridge, we need to turn it on the boolean flag.
- The relevant terraform syntax are as follows.

https://github.com/umccr/infrastructure/blob/4ac93e00e056c356e37007b70b440db7c881e04c/terraform/stacks/data_archive/byob_ica_v2.tf#L511-L515

- Note that we have commented out the TF code. The reason being as follows.

> NOTE:
> S3 Buckets only support a single notification configuration. Declaring multiple aws_s3_bucket_notification resources to the same S3 Bucket will cause a perpetual difference in configuration. See the example "Trigger multiple Lambda functions" for an option.
> https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_notification

- Basically we do not wish to undo nor clash with bucket notification with ICAv2 BYOB integration.
- For now, we simply handle turning on EventBridge bucket notification manually.

**Comment by victorskl** [comment]
Caveat 2:

- Refer to [AWS doc](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-cross-account.html), we are following EventBridge to EventBridge cross account integration.
- There are source account and target account.
- At source account (UoM data account) we configure terraform to forward S3 events to target. These [code block](https://github.com/umccr/infrastructure/blob/4ac93e00e056c356e37007b70b440db7c881e04c/terraform/stacks/data_archive/byob_ica_v2.tf#L508-L553).
- At target account EventBridge, we still have to allow events from the source. This has to configure permission policy at designated target bus. In this case, we are targeting to "default" bus at destination. As follows.

https://github.com/umccr/infrastructure/blob/4ac93e00e056c356e37007b70b440db7c881e04c/terraform/stacks/data_archive/byob_ica_v2.tf#L17

- Since destination bus is AWS EventBridge "default" bus, we configure the permission, manually. As follows.

`AWS Console > EventBridge > Event buses > default > Permissions`

```
{
  "Version": "2012-10-17",
  "Statement": [{
    "Sid": "AllowDataAccountToPutEvents",
    "Effect": "Allow",
    "Principal": {
      "AWS": "arn:aws:iam::<DATA_ACCOUNT_ID>:root"
    },
    "Action": "events:PutEvents",
    "Resource": "arn:aws:events:ap-southeast-2:<DEV_ACCOUNT_ID>:event-bus/default"
  }]
}
```
